File: android/src/main/kotlin/eu/kevin/kevin_flutter/model/KevinBank.kt
package eu.kevin.kevin_flutter.model

import eu.kevin.accounts.bankselection.entities.Bank
import kotlinx.serialization.Serializable

@Serializable
internal data class KevinBank(
    val id: String,
    val name: String,
    val officialName: String? = null,
    val imageUri: String,
    val bic: String? = null,
)

internal fun Bank.toKevinBank() = KevinBank(id, name, officialName, imageUri, bic)

File: packages/kevin_flutter_accounts/ios/Classes/Model/KevinBank.swift
import kevin_ios

internal struct KevinBank: Codable {
    let id: String
    let name: String
    let officialName: String?
    let imageUri: String
    let bic: String?
    
    init(
        id: String,
        name: String,
        officialName: String? = nil,
        imageUri: String,
        bic: String? = nil
    ) {
        self.id = id
        self.name = name
        self.officialName = officialName
        self.imageUri = imageUri
        self.bic = bic
    }
}

internal extension ApiBank {
    func toKevinBank() -> KevinBank {
        return KevinBank(
            id: self.id,
            name: self.name,
            officialName: self.officialName,
            imageUri: self.imageUri,
            bic: self.bic)
    }
}
